core/src/main/java/com/radixdlt/api/system/routes/HealthHandler.java

```diff
@@ -122,24 +130,85 @@ public HealthResponse handleRequest() {
                             readinessSignalStatuses.getOrDefault(
                                 p.protocolUpdateTrigger().nextProtocolVersion(),
-                                NO_SIGNAL_REQUIRED)))
+                                NO_SIGNAL_REQUIRED),
+                            ledgerSummary))
                 .toList());
   }
 
   private com.radixdlt.api.system.generated.models.PendingProtocolUpdate pendingProtocolUpdate(
       ProtocolState.PendingProtocolUpdate pendingProtocolUpdate,
-      PendingProtocolUpdate.ReadinessSignalStatusEnum readinessSignalStatus) {
-    final var state =
-        switch (pendingProtocolUpdate.state()) {
-          case ProtocolState.PendingProtocolUpdateState.Empty
-          empty -> new EmptyPendingProtocolUpdateState().type(PendingProtocolUpdateStateType.EMPTY);
-          case ProtocolState.PendingProtocolUpdateState.ForSignalledReadinessSupportCondition
-          forSignalledReadinessSupportCondition -> new SignalledReadinessPendingProtocolUpdateState()
-              .thresholdsState(
-                  forSignalledReadinessSupportCondition.thresholdsState().stream()
-                      .map(
-                          thresholdState ->
-                              new SignalledReadinessPendingProtocolUpdateStateAllOfThresholdsState()
+      PendingProtocolUpdate.ReadinessSignalStatusEnum readinessSignalStatus,
+      LedgerSummary ledgerSummary) {
+    final var currentEpoch = ledgerSummary.currentEpoch();
+    final var epochStartTimeCalculator =
+        new EpochStartTimeCalculator(
+            ledgerSummary.consensusManagerConfigEpochTargetDurationMs(),
+            currentEpoch,
+            ledgerSummary.consenusManagerStateEpochEffectiveStartMs());
+
+    return switch (pendingProtocolUpdate.protocolUpdateTrigger().enactmentCondition()) {
+      case EnactAtStartOfEpochIfValidatorsReady condition -> {
+        final var state = (ForSignalledReadinessSupportCondition) pendingProtocolUpdate.state();
+
+        final var projectedFulfillmentEpochByThreshold =
+            state.thresholdsState().stream()
+                .collect(
+                    Collectors.toMap(
+                        Tuple.Tuple2::first,
+                        thresholdState -> {
+                          final var consecutiveStartedEpochsOfSupport =
+                              thresholdState.last().consecutiveStartedEpochsOfSupport().toLong();
+                          if (consecutiveStartedEpochsOfSupport <= 0) {
+                            return Option.<Long>empty();
+                          } else {
+                            final var requiredConsecutiveCompletedEpochsOfSupport =
+                                thresholdState
+                                    .first()
+                                    .requiredConsecutiveCompletedEpochsOfSupport()
+                                    .toLong();
+                            final var projectedFulfillmentEpoch =
```

Comment:
To make it easier to verify line-by-line, I'd prefer this to be called `projectedFulfillmentAtStartOfEpoch`